cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/SimpleJob.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule.queue;
+
+import co.cask.cdap.internal.app.runtime.schedule.ProgramSchedule;
+import co.cask.cdap.proto.Notification;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * Simple implementation of {@link Job}.
+ */
+public final class SimpleJob implements Job {
+  private final ProgramSchedule schedule;
+  private final long creationTime;
+  private final List<Notification> notifications;
+  private final State state;
+
+  public SimpleJob(ProgramSchedule schedule, long creationTime, List<Notification> notifications, State state) {
+    this.schedule = schedule;
+    this.creationTime = creationTime;
+    this.notifications = ImmutableList.copyOf(notifications);
+    this.state = state;
+  }
+
+  public ProgramSchedule getSchedule() {
+    return schedule;
+  }
+
+  public long getCreationTime() {
+    return creationTime;
+  }
+
+  public List<Notification> getNotifications() {
+    return notifications;
+  }
+
+  public State getState() {
+    return state;
+  }
+
+  @Override
+  public JobKey getJobKey() {
+    return new JobKey(schedule.getScheduleId(), creationTime);
```

Comment:
Better just have the `JobKey` as a field and remove `schedule` and `creationTime` field since they are already part of the `JobKey. For the getter, you can return `jobKey.getSchedule()` and `jobKey.getCreationTime()`